common/src/main/java/org/apache/comet/vector/CometDictionary.java

```diff
@@ -83,6 +82,18 @@ public byte[] decodeToBinary(int index) {
       case FIXEDSIZEBINARY:
         return values.getBinary(index);
       case DECIMAL:
+        if (binaries == null) {
```

Comment:
I wonder if this need to be made thread safe?

Reply:
I don't think this gets read in multiple threads, but even if it would be, it is overwrite, so should be fine.